src/rust-crypto/verification.ts

```diff
@@ -75,11 +75,15 @@ export class RustVerificationRequest
         const onChange = async (): Promise<void> => {
             const verification: RustSdkCryptoJs.Qr | RustSdkCryptoJs.Sas | undefined = this.inner.getVerification();
 
-            // If we now have a `Verification` where we lacked one before, or we have transitioned from QR to SAS,
-            // wrap the new rust Verification as a js-sdk Verifier.
             if (verification instanceof RustSdkCryptoJs.Sas) {
                 if (this._verifier === undefined || this._verifier instanceof RustQrCodeVerifier) {
+                    // If we now have a `Verification` where we lacked one before, or we have transitioned from QR to SAS,
+                    // wrap the new rust Verification as a js-sdk Verifier.
                     this.setVerifier(new RustSASVerifier(verification, this, outgoingRequestProcessor));
+                } else if (this._verifier instanceof RustSASVerifier) {
+                    // We may get a new rust `Verification` if both parties try to start the verification at the same time,
+                    // but we lose the tie breaking.
+                    this._verifier.setInner(verification);
                 }
```

Comment:
Did you consider replacing the `RustSASVerifier`, rather than updating the existing one? When I designed this, I was rather hoping to avoid having to do this sort of switcheroo, and it feels like a replacement might be simpler.

If it's a case of documenting that clients aren't allowed to hold onto a handle to the verifier, that would be ok I think.

But there might be more fundamental reasons that won't work.

Reply:
I think the client can call `.verify()` on the `RustSASVerifier`, which is supposed to return once the verification succeeds, but that won't work if a new `RustSASVerifier` is used.